Update documentation with PSP removal (#3360)
rancher/rke2: docs/security/hardening_guide.md

Network Policies are namespace scoped. When a network policy is introduced to a given namespace, all traffic not allowed by the policy is denied. However, if there are no network policies in a namespace all traffic will be allowed into and out of the pods in that namespace.

**Remediation**

> This can be remediated by setting `profile: "cis-1.6"` or `cis-1.23` in newer versions in RKE2 configuration file `/etc/rancher/rke2/config.yaml`. An example can be found below.

Reviewer: ```suggestion
This can be remediated by staring RKE2 with the `profile` flag set in the configuration file. An example can be found below.
```

## RKE2 configuration

**Below v1.25.0**

Below is the minimum necessary configuration needed for hardening RKE2 to pass CIS v1.6 hardened profile `rke2-cis-1.6-profile-hardened` available in Rancher.

```yaml
secrets-encryption: "true"
profile: "cis-1.6"           # CIS 4.2.6, 5.2.1, 5.2.8, 5.2.9, 5.3.2
```

> **v1.25.0 and Higher**

Reviewer: Put 1.25 first
